cmake/match.py

```diff
@@ -35,9 +42,26 @@
         else:
             pattern += part
 
-    input_line = input_lines[i].strip()
+    # empty input file or end of input file, from now on match file must be optional
+    if not input_lines:
+        if optional_line is True:
+            continue
+        else:
+            print("End of input file or empty file.")
+            print("expected: " + match_line.strip())
+            sys.exit(1)
+
+    input_line = input_lines[input_idx].strip()
     if not re.fullmatch(pattern, input_line):
-        print(f"Line {i+1} does not match".format(i+1))
-        print("is:       " + input_line)
-        print("expected: " + match_line.strip())
-        sys.exit(1)
+        if optional_line is True:
+            continue
+        else:
+            print(f"Line {i+1} does not match".format(i+1))
```

Comment:
hmm maybe I'm mistaken, but it seems you're mixing F-String with `str.format()` syntax 😉 either one or another...?

Reply:
Done.